Description: Workaround for a solved typing bug in IcedTea javac compiler
 For a long time the javac had a bug with the handling of wildcard types
 that results in a hole in the type system. Since IcedTea 1.8.2 the bug is
 solved but it broke some code in this package. For more info, please
 visit: http://bugs.sun.com/bugdatabase/view_bug.do?bug_id=6638712
Author: Miguel Landaeta <miguel@miguel.cc>
Bug-Debian: http://bugs.debian.org/602362
Forwarded: not-needed
Last-Update: 2011-10-18

--- libjsr166y-java-1.7.0.orig/src/extra166y/AbstractParallelAnyArray.java
+++ libjsr166y-java-1.7.0/src/extra166y/AbstractParallelAnyArray.java
@@ -1166,7 +1166,7 @@ public abstract class AbstractParallelAn
             (IntAndObjectPredicate<? super T> selector) {
             return new ORPap<T>
                 (ex, origin, fence, array,
-                 compoundIndexedSelector(this.selector, selector));
+                 compoundIndexedSelector((IntAndObjectPredicate)this.selector, selector));
         }
 
         public <U> ParallelArrayWithMapping<T, U> withMapping
@@ -1700,14 +1700,14 @@ public abstract class AbstractParallelAn
             (Predicate<? super T> selector) {
             return new ORPap<T>
                 (ex, origin, fence, array,
-                 compoundIndexedSelector(this.selector, selector));
+                 compoundIndexedSelector((IntAndObjectPredicate)this.selector, selector));
         }
 
         public ParallelArrayWithFilter<T> withIndexedFilter
             (IntAndObjectPredicate<? super T> selector) {
             return new ORPap<T>
                 (ex, origin, fence, array,
-                 compoundIndexedSelector(this.selector, selector));
+                 compoundIndexedSelector((IntAndObjectPredicate)this.selector, selector));
         }
 
         public <U> ParallelArrayWithMapping<T, U> withMapping
